### DISCUSSION TITLE
Using Budibase as a Slack bot framework?

### DISCUSSION THREAD

**Discussion by Nezteb** [Budibase/budibase#4648] [Help] [answered]
I’m building a Slack bot that I want to work like so:

- The bot has one of a dozen commands with specific inputs
- A service (maybe Budibase?) does some work involving databases, cron jobs, running scripts, and more based on some state and the command/inputs
- The status/results of these job(s) can be viewed and interacted with via web GUI (also Budibase?)
- Bot reports the results of the job(s) once finished

The Budibase docs explicitly mention Budibase-to-Slack integration but not the other way around.

I assume this could be possible by using Budibase webhooks and Slack events?

- https://budibase.com/integration/webhook/
- https://api.slack.com/apis/connections/events-api

I’m going to try it out and see, but I’m curious if anyone else has set up bi-directional communication between Budibase and another third-party service like this?

**Comment by mjashanks** [comment, marked as answer]
In the next few weeks, we are releasing official support for Budibase API. So you'll be able to do CRUD operations on all your datasources - this will probably help you as well.

**Reply by Nezteb** [reply]
Hi @mjashanks! I had a reminder in my phone to check on this after 3 weeks. Have there been any updates? If not, no worries! I just wanted to check. 😄

**Reply by mjashanks** [reply]
Yes! The budibase public API has been released. 

Start here: https://docs.budibase.com/docs/public-api
Reference here: https://docs.budibase.com/reference/post_applications

**Reply by Nezteb** [reply]
@mjashanks Awesome!

Quick related question: Can Budibase Actions be HTTP calls? Example workflow I'm imagining:

1. Use invokes Slack bot via slash command.
2. Slack bot makes HTTP call to Budibase. I don't know if this would use the API or a Budibase webhook; the API doesn't seem to have a way to trigger automations.
3. Budibase triggers automation.
4. Automation actions makes several HTTP calls to various services like CI, Kubernetes, other third-party APIs, etc. I'm not sure if Budibase currently supports HTTP calls as an action; the docs suggest not.
5. Once those services are done, they call another Budibase webhook (assuming the services support webhooks).
6. Budibase either sends results in Slack message or makes another HTTP call to the Slack bot itself to handle the messages.

Does that seem doable with Budibase today?
